lib/wakes/location_string.rb

```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+module LocationString
```

Comment:
@benhutton What do you think about this syntax for `RedirectMapper.initialize`

```
      @source_host, @source_path = URIFromLocationString.get_host_and_path(source_path_or_url)
      @target_host, @target_path = URIFromLocationString.get_host_and_path(target_path_or_url)
```

and this

```
class URIFromLocationString
  def self.generate(location_string)
    location_string = "http://#{location_string}" if URI(location_string).scheme.nil?
    URI(location_string)
  end

  def self.get_host_and_path(location_string)
    uri = self.generate(location_string)
    uri.host, uri.path
  end
end
```